src/dotty/tools/dotc/core/tasty/NameBuffer.scala

```diff
@@ -40,13 +41,12 @@ class NameBuffer extends TastyBuffer(10000) {
       nameIndex(name)
   }
 
-  private def withLength(op: => Unit): Unit = {
+  private def withLength(op: => Unit, lengthWidth: Int = 1): Unit = {
     val lengthAddr = currentAddr
-    writeByte(0)
+    for (i <- 0 until lengthWidth) writeByte(0)
     op
     val length = currentAddr.index - lengthAddr.index - 1
-    assert(length < 128)
-    putNat(lengthAddr, length, 1)
+    putNat(lengthAddr, length, lengthWidth)
```

Comment:
There's already an assertion in NameBuffer that we don't overflow, and which even comes with a comprehensible error message. So we are good.